charm-slurmctld/src/etcd_ops.py

```diff
@@ -0,0 +1,100 @@
+"""etcd operations."""
+
+import logging
+import shlex
+import shutil
+import subprocess
+import tarfile
+import os
+from tempfile import TemporaryDirectory
+from pathlib import Path
+from typing import List
+
+from etcd3gw.client import Etcd3Client
+
+logger = logging.getLogger()
+
+
+class EtcdOps:
+    """ETCD ops."""
+
+    def __init__(self):
+        """Initialize class."""
+        # system user and group
+        self._etcd_user = "etcd"
+        self._etcd_group = "etcd"
+        self._etcd_service = "etcd.service"
+
+    def install(self, resource_path: Path):
+        """Install etcd."""
+        # extract resource tarball
+        with TemporaryDirectory(prefix="omni") as tmp_dir:
+            logger.debug(f"## extracting {resource_path} to {tmp_dir}")
+            with tarfile.open(resource_path, 'r') as tar:
+                tar.extractall(path=tmp_dir)
+
+            usrbin = Path("/usr/bin")
+            logger.debug(f"## installing etcd in {usrbin}")
+            source = Path(tmp_dir) / "etcd-v3.5.0-linux-amd64"
+            for abin in ["etcd", "etcdutl", "etcdctl"]:
+                shutil.copy2(source / abin, usrbin / abin)
+
+        self._create_etcd_user_group()
+
+        varlib = Path("/var/lib/etcd")
+        if not varlib.exists():
+            varlib.mkdir()
+        shutil.chown(varlib, user=self._etcd_user, group=self._etcd_group)
+
+        self._setup_systemd()
+
+    def _create_etcd_user_group(self):
+        logger.debug("## creating etcd user and group")
+        cmd = f"groupadd {self._etcd_group}"
+        subprocess.call(shlex.split(cmd))
+
+        cmd = f"useradd --system --no-create-home --gid {self._etcd_group} " \
+                      f"--shell /usr/sbin/nologin {self._etcd_user}"
+        subprocess.call(shlex.split(cmd))
+
+    def _setup_systemd(self):
+        logger.debug("## creating systemd files for etcd")
+
+        charm_dir = os.path.dirname(os.path.abspath(__file__))
```

Comment:
Since you are using pathlib Path on the next line, I think it might make sense to just use pathlib for this as well:

```python
charm_dir = Path(__file__).parent
```